GraphSturcture/src/graphStructure/Graph.java

```diff
@@ -0,0 +1,192 @@
+package graphStructure;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+
+public class Graph {
+
+  private Map<Integer, Vertex> vertices;
+  private Map<Integer, Edge> edges;
+  private Queue<Integer> unusedVertexId;// to store the id which is not used,"for example ,when a vertex is deleted"
```

Comment:
Hi, I want to maintain the id of vertices, because,for example a graph G which has 5 vertices"5",now when a new vertex is deleted, for example , vertex 3 is deleted. And then we add a new vertex into the graph G, there is two choice to allocate an id to the new vertex, 3 or 6. So if we maintain a queue of unusedvertexId we can get the first unused id to the new vertex. Otherwise, we should waste some time to search the id which have not been used ,and allocate the id to the vertex.

Reply:
Do not try to make things complicated. Most of the time, we will create a graph by adding edges directly into it, and while it is loaded, we will just run algorithm without considering modify it unless the algorithm need that. Programming is like building a skycrapper, you always start with the easiest task, and add bricks and decorations when there is a need. Keeping it simple can avoid mistakes at the beginning, and at the same time, make me more focused on your coding format and the other things without too much distraction.